spec/models/article_spec.rb

```diff
@@ -62,5 +63,34 @@
 
       Article.cached_all.should eq [article]
     end
+
+    it "should correct key cache show" do
+      article = FactoryGirl.create(:article)
+      Rails.cache.should_receive(:fetch).with(['Article', article.id]).once
+      Article.cached_find(article.id)
+    end
+
+    it "should get article by id" do
+      article = FactoryGirl.create(:article)
+      find = Article.cached_find(article.id)
+      expect(find.id).to eq(article.id)
+    end
+
+    it 'should invalidate cached_find on upadte' do
+      article = FactoryGirl.create(:article)
+      find = Article.cached_find(article.id)
+
+      article.title = 'Hahaa'
+      article.save!
+
+      expect(Article.cached_find(article.id)).to eq(article)
+    end
+
+    it "should get article comment from cached" do
+      article = FactoryGirl.create(:article)
+      comment = FactoryGirl.create(:comment, article_id: article.id)
+
+      expect(article.cached_comments.count).to eq(1)
```

Comment:
@phendych this test is useless if you are testing the cache, you are missing something.